lfs.c

```diff
@@ -2418,6 +2470,10 @@ int lfs_file_opencfg(lfs_t *lfs, lfs_file_t *file,
     lfs->mlist = (struct lfs_mlist*)file;
 
     if (tag == LFS_ERR_NOENT) {
+#ifdef LFS_READONLY
+        err = LFS_ERR_NOENT;
+        goto cleanup;
+#else
```

Comment:
I just have a minor nit here, since the `#ifdef` starts inside an if statement and ends outside. It's a complex location for LFS_READONLY, but maybe this would be better this way?

``` c
#ifdef LFS_READONLY
    if (tag == LFS_ERR_NOENT) {
        err = LFS_ERR_NOENT;
        goto cleanup;
#else
    if (tag == LFS_ERR_NOENT) {
        if (!(flags & LFS_O_CREAT)) {
            err = LFS_ERR_NOENT;
            goto cleanup;
        }
        ... snip ...
    } else if (flags & LFS_O_EXCL) {
        err = LFS_ERR_EXIST;
        goto cleanup;
#endif
    } else if (lfs_tag_type3(tag) != LFS_TYPE_REG) {
        err = LFS_ERR_ISDIR;
        goto cleanup;
        ... snip ...
    }
```

Other than that these changes are looking great.

Reply:
94c9c4c should do the trick!